lib/db/lowlevel.go

```diff
@@ -7,88 +7,93 @@
 package db
 
 import (
-	"os"
-	"strings"
-	"sync"
+	"bytes"
+	"encoding/binary"
+	"fmt"
 	"sync/atomic"
 
-	"github.com/syndtr/goleveldb/leveldb"
-	"github.com/syndtr/goleveldb/leveldb/errors"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
-	"github.com/syndtr/goleveldb/leveldb/opt"
-	"github.com/syndtr/goleveldb/leveldb/storage"
 	"github.com/syndtr/goleveldb/leveldb/util"
-)
 
-const (
-	dbMaxOpenFiles = 100
-	dbWriteBuffer  = 16 << 20
-	dbFlushBatch   = dbWriteBuffer / 4 // Some leeway for any leveldb in-memory optimizations
+	"github.com/syncthing/syncthing/lib/protocol"
 )
 
+// Backend is the interface that a database needs to implement to be used with
+// Syncthing.
+// It is not guaranteed that after calling Close no more calls are made to other
+// methods (see ErrorChecker.IsClosed).
+type Backend interface {
+	ErrorChecker
+	Get(key []byte) ([]byte, error)
+	Has(key []byte) (bool, error)
+	Put(key, val []byte) error
+	Delete(key []byte) error
+	NewBatch() Batch
+	NewIterator(slice *util.Range) (iterator.Iterator, error)
```

Comment:
I thought as this are packages separate from the meat of the golevel implementation it could be fine to use them, but sure makes sense to have the full interface/types here. As it is plainly at least inspired by syndtr/goleveldb, I probably need to attribute that somehow? Maybe a comment at the top referencing to `LICENSE-syndtr` which contains the copyright (BSD 2-clause, which is fine with MPL I think). Or am I overthinking this?